doc/guides/modules-in-core.md

```diff
@@ -0,0 +1,44 @@
+# To be or not to be in core
+
+Should a module should be in core? This question arises every so often. This
+document explains things to consider when deciding whether a module should be in
+core or not.
+
+## Strong arguments for including a module in core
+
+1. The module provides functionality that is standardized (such as a
+   [Web API][]) and overlaps with existing functionality.
+2. The module can only be implemented in core.
+3. The module can only be implemented in a performant way in core.
+4. Developer experience is significantly improved if the module is in core.
+5. The module will be widely used, have cross platform native code, and being in
```

Comment:
My sense is that while it is not always possible to know, surely there are cases where we can be certain. For example, a core implementation of `fetch` would certainly be widely used.  But this bit of text you're asking about is me paraphrasing (or maybe even just copying) something @mhdawson wrote so maybe he would like to respond.

Reply:
That's what I mean though, how can we be certain `fetch` would be widely used? Just because it's a web API? Because of some vague npm stats? Something else? What if people end up preferring/using a more flexible API (perhaps based on the existing built-in `http*` modules) instead?

I feel like it's easy to mistake the chanting of a vocal minority as some kind of guarantee the usefulness of a module in core.